src/libponyc/pass/docgen.c

```diff
@@ -403,6 +407,30 @@ static void doc_type_list(docgen_t* docgen, ast_t* list, const char* preamble,
   fprintf(docgen->type_file, "%s", postamble);
 }
 
+static bool is_for_testing(const char* name, ast_t* list)
+{
+  assert(name != NULL);
+  assert(list != NULL);
+
+  if (strncmp(name, "_Test", 5) == 0) return true;
+
+  if(ast_id(list) == TK_NONE)
+    return false;
+
+  for(ast_t* p = ast_child(list); p != NULL; p = ast_sibling(p))
+  {
+    if (ast_id(p) == TK_NOMINAL)
+    {
+      AST_GET_CHILDREN(p, package, id, tparams, cap, ephemeral);
```

Comment:
The `id` is the second child of a `TK_NOMINAL`. So the `AST_GET_CHILDREN` you have is fine. You could stop at `id` (like your example above) as well. You could also do:

```c
ast_t* id = ast_childidx(p, 1);
```